File: src/main/java/com/qlik/demo/java/model/SeatOption.java
package com.qlik.demo.java.model;

import java.util.Arrays;
import java.util.Optional;

public enum SeatOption {
    ECONOMY("Economy"),
    FIRST_CLASS("First Class");

    private final String optionName;

    SeatOption(String optionName) {
        this.optionName = optionName;
    }

    // Reverse lookup
    public static Optional<SeatOption> get(final String name) {
        return Arrays.stream(SeatOption.values())
                .filter(seatClass -> seatClass.optionName.equals(name))
                .findFirst();
    }
}

File: src/main/kotlin/com/qlik/demo/kotlin/model/SeatOption.kt
package com.qlik.demo.kotlin.model

enum class SeatOption(val optionName: String) {
    FIRST_CLASS("First Class"),
    ECONOMY("Economy");

    companion object {
        fun from(optionName: String): SeatOption? =
            entries.firstOrNull { it.optionName == optionName }
    }
}